### ISSUE TITLE
Cannot bind to port 53

### ISSUE THREAD

**Issue by LewisSpring** [AdguardTeam/AdGuardHome#3760]
### Issue Details

* **Version of AdGuard Home server:**
  * v0.106.3
* **How did you install AdGuard Home:**
  * Docker
* **How did you setup DNS configuration:**
  * System
* **If it's a router or IoT, please write device model:**
  * N/A
* **CPU architecture:**
  * AMD64
* **Operating system and version:**
  * Debian 11

### Expected Behavior
Docker Container starts and begins to listen for DNS requests.

### Actual Behavior
Docker container boot loops, 

```
2021/10/21 23:25:59.032947 [info] AdGuard Home, version v0.106.3

2021/10/21 23:25:59.092399 [info] Initializing auth module: /opt/adguardhome/work/data/sessions.db

2021/10/21 23:25:59.092532 [info] auth: initialized.  users:1  sessions:2

2021/10/21 23:25:59.092997 [info] number of certs - 3

2021/10/21 23:25:59.093013 [info] got an intermediate cert

2021/10/21 23:25:59.093024 [info] got an intermediate cert

2021/10/21 23:25:59.131681 [info] Initialize web module

2021/10/21 23:25:59.143471 [info] AdGuard Home is available on the following addresses:

2021/10/21 23:25:59.143542 [info] AdGuard Home is available on the following addresses:

2021/10/21 23:25:59.143771 [info] Go to https://127.0.0.1:9001

2021/10/21 23:25:59.143781 [info] Go to https://172.17.0.3:9001

2021/10/21 23:25:59.143832 [info] Go to http://127.0.0.1:9001

2021/10/21 23:25:59.143893 [info] Go to http://172.17.0.3:9001

2021/10/21 23:25:59.305598 [info] Starting the DNS proxy server

2021/10/21 23:25:59.305617 [info] Ratelimit is enabled and set to 20 rps

2021/10/21 23:25:59.305621 [info] The server is configured to refuse ANY requests

2021/10/21 23:25:59.305624 [info] DNS cache is enabled

2021/10/21 23:25:59.305631 [info] MaxGoroutines is set to 300

2021/10/21 23:25:59.305638 [info] Creating the UDP server socket

2021/10/21 23:25:59.375250 [fatal] couldn't start forwarding DNS server: couldn't listen to UDP socket, cause: listen udp 172.17.0.7:53: bind: cannot assign requested address

```

### Additional Information
Sometimes, manually restarting the container will work.

Ports:

| Host | Container | Type | Notes |
| ------------- | ------------- | ------------- | ------------- |
| 53  | 53  | TCP |
| 9443  | 443  | TCP |
| 853 | 853 | TCP |
| 3000 | 3000 | TCP |
| 4000 | 4000 | TCP |
| 9001 | 9001 | TCP| UI Port |
| 53 | 53 | UDP |
| 9443 | 443 | UDP |

Networking is set to Bridge mode to avoid port conflicts.

Cheers :)

**Comment by LewisSpring** [comment]
Hi all,
The issue still remains - even if the system has been booted for some time.

**Comment by LewisSpring** [comment]
After rebooting, there's no programs claiming to use :53, but yet AdGuard can;t bind still.

**Comment by ainar-g** [comment]
Unfortunately, we still have no idea what could be wrong there.  Could you please show the full output of `ifconfig` from within the container?

**Comment by LewisSpring** [comment]
Hello Ainar, 
A few days ago I rebuilt the container, and haven't had any issues with binding to port 53.
I'm using bridge mode. Would it be more stable if I used host instead? I'm not using any ports like 80 or 443 in the container so there shouldn't be an port conflicts on my host.

IFConfig:

```
eth0      Link encap:Ethernet  HWaddr 02:42:AC:11:00:03
          inet addr:172.17.0.3  Bcast:172.17.255.255  Mask:255.255.0.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:693 errors:0 dropped:0 overruns:0 frame:0
          TX packets:730 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:159480 (155.7 KiB)  TX bytes:159392 (155.6 KiB)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

```

**Comment by ainar-g** [comment]
I can see that in your `ifconfig` output, the IP is `172.17.0.3`, while the address in the original error message is `172.17.0.7:53`.  What is the value of your `bind_host`?

**Comment by LewisSpring** [comment]
That could be related to the recreation of the container.
Bind Host of new container is 0.0.0.0.
Old container had the same IP.

**Comment by ainar-g** [comment]
Unfortunately, we have no other ideas.  I'll leave this issue open with a help wanted label so that people with similar issues could chime in.

**Comment by LewisSpring** [comment]
Hi Ainar,
As of recent, This issue still seems to have not come back.
Not sure what the issue was, but it still seems stable right now.
Thank you

**Comment by SDS1337** [comment]
I seem to have had this exact issue and managed to find the cause.
In `AdGuardHome.yaml`, 
```yaml
dns:
  bind_hosts:
  - 172.20.0.11
```
The error I was getting was
> couldn't start forwarding DNS server: couldn't listen to UDP socket, cause: listen udp 172.20.0.11:53: bind: cannot assign requested address

I noticed that `172.20.0.11` was the incorrect network interface so simply changing the config entry to
```yaml
dns:
  bind_hosts:
  - 0.0.0.0
```
seems to have fixed the issue. 
Not sure what caused it. I don't think I chose a specific interface for the DNS to listen on during setup, since it was working fine until a server restart today.
